models/DataObject/ClassDefinition.php

```diff
@@ -731,6 +734,17 @@ protected function deletePhpClasses()
         @unlink($this->getDefinitionFile());
     }
 
+
+    public function isWritable()
+    {
+        $name = $this->getName();
+        $customFile = PIMCORE_CUSTOM_CONFIGURATION_DIRECTORY . '/classes/definition_' . $name . '.php';
```

Comment:
I'd go for the following here: 
```php
return (strpos($this->getDefinitionFile(), PIMCORE_CUSTOM_CONFIGURATION_DIRECTORY) === 0);
```
This has the benefit that we have the way how the path is being build is just in one place. 
Of course this should be done for the other classes as well.

Reply:
Since we're on PHP 8, `str_starts_with` should be used:
```php
return str_starts_with($this->getDefinitionFile(), PIMCORE_CUSTOM_CONFIGURATION_DIRECTORY);
```

Reply:
Absolutely 👍 Thanks! 😊